qiskit/primitives/base_sampler.py

```diff
@@ -209,6 +215,27 @@ def parameters(self) -> tuple[ParameterView, ...]:
         """
         return tuple(self._parameters)
 
+    @property
+    def run_options(self) -> Options:
+        """Return options values for the estimator.
+
+        Returns:
+            run_options
+        """
+        return self._run_options
+
+    def set_run_options(self, **fields) -> BaseSampler:
+        """Set options values for the estimator.
+
+        Args:
+            **fields: The fields to update the options
+
+        Returns:
+            self
+        """
+        self._run_options.update_options(**fields)
+        return self
```

Comment:
Does setter need to return `self`? Do you think of a chain operation `sampler.set_run_options(...).run(...)`?

Reply:
Thanks... i didn't think so deeply...
I'll remove it if there is no use case.

Reply:
I would also not expect this to return anything as it's called "set_..." 🙂